testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replicator_encryption_hook.py

```diff
@@ -0,0 +1,107 @@
+import pytest
+import time
+
+from keywords.MobileRestClient import MobileRestClient
+# from keywords.ClusterKeywords import ClusterKeywords
+# from keywords import couchbaseserver
+# from keywords.utils import log_info, random_string, get_embedded_asset_file_path
+# from CBLClient.Database import Database
+from CBLClient.Replication import Replication
+from CBLClient.Document import Document
+# from CBLClient.Authenticator import Authenticator
+# from concurrent.futures import ThreadPoolExecutor
+# from CBLClient.Blob import Blob
+# from CBLClient.Dictionary import Dictionary
+# from libraries.testkit.prometheus import verify_stat_on_prometheus
+# from keywords.SyncGateway import sync_gateway_config_path_for_mode
+from keywords import document
+from libraries.testkit import cluster
+# from utilities.cluster_config_utils import persist_cluster_config_environment_prop, copy_to_temp_conf
+# from keywords.attachment import generate_2_png_100_100
+# from keywords.SyncGateway import SyncGateway
+
+
+@pytest.mark.listener
+@pytest.mark.parametrize("type", [
+    ("push")
+])
+def test_replication_with_encryption(params_from_base_test_setup, type):
+    """
+    @summary:
+    1. Have SG and CBL up and running
+    2. Create a simple document with encryption property
+    3. Start the replicator and make sure documents are
+    replicated on SG
+
+    7. Update the normal data
```

Comment:
summary need to be up-to-date to represent what the test case is doing, if not, it will be hard to future investigation.